File: iplookup.rs
extern crate reqwest;
use reqwest::header;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut headers = header::HeaderMap::new();
    headers.insert("authority", "tronghoa.dev".parse().unwrap());
    headers.insert("accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9".parse().unwrap());
    headers.insert("accept-language", "en-US,en;q=0.9".parse().unwrap());
    headers.insert("referer", "https://tronghoa.dev/".parse().unwrap());
    headers.insert("sec-ch-ua", "\" Not;A Brand\";v=\"99\", \"Microsoft Edge\";v=\"103\", \"Chromium\";v=\"103\"".parse().unwrap());
    headers.insert("sec-ch-ua-mobile", "?0".parse().unwrap());
    headers.insert("sec-ch-ua-platform", "\"Windows\"".parse().unwrap());
    headers.insert("sec-fetch-dest", "document".parse().unwrap());
    headers.insert("sec-fetch-mode", "navigate".parse().unwrap());
    headers.insert("sec-fetch-site", "same-origin".parse().unwrap());
    headers.insert("sec-fetch-user", "?1".parse().unwrap());
    headers.insert("upgrade-insecure-requests", "1".parse().unwrap());
    headers.insert("user-agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36 Edg/103.0.1264.37".parse().unwrap());

    let client = reqwest::blocking::Client::new();
    let res = client.get("https://tronghoa.dev/api/ip_lookup.php?ipadress=me")
        .headers(headers)
        .send()?
        .text()?;
    println!("{}", res);

    Ok(())
}

File: iplookup.R
require(httr)

headers = c(
  `authority` = 'tronghoa.dev',
  `accept` = 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
  `accept-language` = 'en-US,en;q=0.9',
  `referer` = 'https://tronghoa.dev/',
  `sec-ch-ua` = '" Not;A Brand";v="99", "Microsoft Edge";v="103", "Chromium";v="103"',
  `sec-ch-ua-mobile` = '?0',
  `sec-ch-ua-platform` = '"Windows"',
  `sec-fetch-dest` = 'document',
  `sec-fetch-mode` = 'navigate',
  `sec-fetch-site` = 'same-origin',
  `sec-fetch-user` = '?1',
  `upgrade-insecure-requests` = '1',
  `user-agent` = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.53 Safari/537.36 Edg/103.0.1264.37'
)

params = list(
  `ipadress` = 'me'
)

res <- httr::GET(url = 'https://tronghoa.dev/api/ip_lookup.php', httr::add_headers(.headers=headers), query = params)
